internal/libs/autofile/autofile.go

```diff
@@ -78,12 +79,17 @@ func OpenAutoFile(path string) (*AutoFile, error) {
 	af.hupc = make(chan os.Signal, 1)
 	signal.Notify(af.hupc, syscall.SIGHUP)
 	go func() {
-		for range af.hupc {
-			_ = af.closeFile()
+		for {
+			select {
+			case <-af.hupc:
+				_ = af.closeFile()
+			case <-ctx.Done():
+				return
```

Comment:
should we not also closeFile on this branch?

Reply:
I think this is ok, because the `clsoeFileRoutine` calls closeFile when the context cancels, and this is just the signal handler for the sighup listener, which I think is fine if we don't also close here.